server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -1323,7 +1299,7 @@ public boolean containsShard(ShardRouting shard) {
         }
     }
 
-    static final class NodeSorter extends IntroSorter {
+    public static final class NodeSorter extends IntroSorter {
```

Comment:
annoyingly these need to become public so the `StatelessBalancingWeightsFactory` can see/instantiate them